internal/venice-client-common/src/test/java/com/linkedin/venice/utils/RetryUtilsTest.java

```diff
@@ -266,4 +266,36 @@ public void testExponentialBackoffDelayRetryOnSupplier() {
     verify(obj, times(3)).getAnInteger();
     reset(obj);
   }
+
+  @Test
+  public void testFixAttemptDurationOnSupplier() {
+    SomeObj obj = mock(SomeObj.class);
+
+    // Case 1: no failure
+    when(obj.getAnInteger()).thenReturn(1);
+    Assert.assertEquals(
+        (int) RetryUtils.executeWithMaxRetriesAndFixedAttemptDuration(
+            () -> obj.getAnInteger() + 1,
+            3,
+            Duration.ofMillis(100),
+            Collections.singletonList(IllegalStateException.class)),
+        2);
+    verify(obj, times(1)).getAnInteger();
+    reset(obj);
+
+    // Case 2: succeed on the last attempt
+    when(obj.getAnInteger()).thenThrow(IllegalArgumentException.class)
+        .thenThrow(IllegalStateException.class)
+        .thenReturn(2);
+
+    Assert.assertEquals(
+        (int) RetryUtils.executeWithMaxRetriesAndFixedAttemptDuration(
+            () -> obj.getAnInteger() + 1,
+            2,
+            Duration.ofMillis(100),
+            Arrays.asList(IllegalStateException.class, IllegalArgumentException.class)),
+        3);
+    verify(obj, times(3)).getAnInteger();
```

Comment:
can we add some sort of check that it waited for at least 2* `durationPerAttempt` to finish 3 attempts? probably also by providing a bigger `durationPerAttempt`.

Reply:
Hmm what do you mean by this? If it is 3rd attempt then it should have waited 2*durationPerAttempt? Are you saying that the execution time might be > 100ms so no wait in between?

Reply:
Added the timeout to 1s

Reply:
I meant to add a time check before and after `executeWithMaxRetriesAndFixedAttemptDuration` to measure that the total time taken is > 2 secs here? Some way to measure that it did wait for 1secs (including the execution time) between attempts.